app/forms/declaration/BorderTransport.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.declaration
+
+import play.api.data.Forms.{mapping, optional, text}
+import utils.validators.forms.FieldValidator.{isAlphanumeric, isContainedIn, isEmpty, noLongerThan}
+import TransportCodes._
+import play.api.libs.json.Json
+import utils.validators.forms.FieldValidator._
+
+case class BorderTransport(
+  borderModeOfTransportCode: String,
+  meansOfTransportOnDepartureType: String,
+  meansOfTransportOnDepartureIDNumber: Option[String]
+)
+object BorderTransport {
+  val formId = "BorderTransport"
+
+  implicit val formats = Json.format[BorderTransport]
+
+  val formMapping = mapping(
+    "borderModeOfTransportCode" -> optional(
+      text()
+        .verifying(
+          "supplementary.transportInfo.borderTransportMode.error.incorrect",
+          isEmpty or isContainedIn(allowedModeOfTransportCodes)
+        )
+    ).verifying("supplementary.transportInfo.borderTransportMode.error.empty", _.isDefined)
+      .transform[String](
+        optValue => optValue.getOrElse(""),
+        borderModeOfTransportCode => Some(borderModeOfTransportCode)
+      ),
+    "meansOfTransportOnDepartureType" -> optional(
+      text()
+        .verifying(
+          "supplementary.transportInfo.meansOfTransport.departure.error.incorrect",
+          isEmpty or isContainedIn(allowedMeansOfTransportTypeCodes)
+        )
+    ).verifying("supplementary.transportInfo.meansOfTransport.departure.error.empty", _.isDefined)
+      .transform[String](
+        optValue => optValue.getOrElse(""),
+        meansOfTransportOnDepartureType => Some(meansOfTransportOnDepartureType)
+      ),
+    "meansOfTransportOnDepartureIDNumber" -> optional(
+      text()
+        .verifying("supplementary.transportInfo.meansOfTransport.idNumber.error.length", noLongerThan(27))
+        .verifying("supplementary.transportInfo.meansOfTransport.idNumber.error.specialCharacters", isAlphanumeric)
+    )
+  )(BorderTransport.apply)(BorderTransport.unapply)
+}
+
+object TransportCodes {
+
+  val Maritime = "1"
```

Comment:
Still I'm thinking that this will be duplication with Warehouse page. I didn't see rest of the PR yet, so I don't know that you removed it from different place or not.

Reply:
it has been removed